pkg/proxy/switch.go

```diff
@@ -425,6 +417,17 @@ func (s *SwitchSession) Bridge(userConn UserConnection, srvConn srvconn.ServerCo
 		lastActiveTime = time.Now()
 	}
 }
+func (s *SwitchSession) disconnection(room *exchange.Room, parser *Parser, replayRecorder *ReplyRecorder, msg string) {
+	msg = utils.WrapperWarn(msg)
+	replayRecorder.Record([]byte(msg))
+
+	roomMessage := &exchange.RoomMessage{Event: exchange.DataEvent, Body: []byte("\n\r" + msg)}
+	if parser.zmodemParser.IsStartSession() {
+		roomMessage.Body = append(zmodem.SkipSequence, zmodem.CancelSequence...)
+	}
+
+	room.Broadcast(roomMessage)
+}
 
 func (s *SwitchSession) recordSessionFinished(reason model.SessionLifecycleReasonErr) {
 	logObj := model.SessionLifecycleLog{Reason: string(reason)}
```

Comment:
There aren't any apparent errors or deviations that need addressing. The function looks fine from an external perspective, with only minor adjustments needed to adapt it for testing purposes. The implementation appears correct.